### PR TITLE
[Core] Support tensor prompt cache keys for verl-omni

### PR DESCRIPTION
**Pull request by Sky-Trigger**
## Purpose

The prompt embedding cache currently bypasses `encode_prompt` calls whenever
any argument is a tensor. This prevents cache reuse for downstream rollout
integrations such as verl-omni, which pass pre-tokenized prompts as tensors.
In the current verl-omni integration, prompt IDs use `torch.int64`, while
attention masks use either `torch.int64` or `torch.bool`.

This PR:

- Adds cache-key support for strided `torch.int64` and `torch.bool` tensors,
  including tensors on accelerator devices.
- Builds exact tensor keys from dtype, shape, device, and contiguous content
  bytes. Device tensor content is copied directly to CPU bytes for hashing.
- Allows different tensor objects with identical content to share the same
  prompt embedding cache entry.
- Avoids converting tensor values to Python lists.
- Keeps floating-point, meta, and non-strided tensors on the existing
  cache-bypass path.
- Preserves the existing bypass behavior for precomputed prompt embeddings.

## Test Plan

Added unit coverage for:

- `torch.int64` token ID tensors.
- `torch.bool` attention-mask tensors.
- Cache hits across cloned tensors with identical values.
- Cache misses when tensor content or shape changes.
- Cache bypass for unsupported floating-point tensors.
- Cache bypass for precomputed prompt embeddings.

Static checks:

```bash
ruff format --check \
  vllm_omni/diffusion/cache/prompt_embed_cache.py \
  tests/diffusion/cache/test_prompt_embed_cache.py

ruff check \
  vllm_omni/diffusion/cache/prompt_embed_cache.py \
  tests/diffusion/cache/test_prompt_embed_cache.py

git diff --check
```

Focused unit-test command:

```bash
python3 -m pytest -q tests/diffusion/cache/test_prompt_embed_cache.py
```

**vLLM Version:** `0.24.0` (the current verl-omni compatibility target)

**vLLM-Omni Commit:** `485f1be246b5a6498ddd72ce123b08956b50e1a1`  
Rebased onto `main` at `109942465fa4d8e140f9ccac3430bfd90b7cf6de`.

## Test Result

- `ruff format --check`: passed.
- `ruff check`: passed.
- `git diff --check`: passed.
- Focused CPU smoke test: passed.
  - Identical cloned token tensors produced a cache hit.
  - Different token values produced different cache keys.
  - Floating-point tensors continued to bypass the cache.
- The focused pytest command was not completed locally because the current
  Python environment does not have `pytest` installed (`No module named
  pytest`). The corresponding unit tests are included for CI execution.

**BEFORE SUBMITTING:** read
[`CONTRIBUTING.md`](https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md)
and run the
[`precheck-pr` skill](https://github.com/vllm-project/vllm-omni/blob/main/.claude/skills/precheck-pr/SKILL.md)
with the code agent for a self-check against project conventions.

### CODE REVIEW

**Comment by Sky-Trigger** [comment]
@SamitHuang pls take a look.

File: vllm_omni/diffusion/cache/prompt_embed_cache.py [hunk 1]
```diff
@@ -67,14 +67,44 @@
...
+
+    Shape, dtype, and device are part of the key. The contiguous CPU bytes make
+    different tensor objects with the same content share a cache entry without
+    converting token values to Python lists.
+    """
+    if obj.dtype not in _HASHABLE_TENSOR_DTYPES or obj.layout != torch.strided or obj.device.type == "meta":
+        return _NOT_HASHABLE
+
+    try:
+        content = obj.detach().to(device="cpu").contiguous().numpy().tobytes()
```
Comment (SamitHuang) [inline vllm_omni/diffusion/cache/prompt_embed_cache.py:88]: Calling `.to(device="cpu")` on a GPU tensor triggers a host-device synchronization, which will block the CPU thread until the GPU finishes producing this tensor. In highly asynchronous systems like `vLLM-Omni`, this can introduce GPU bubbles. Where are the performance measurements for this sync in the `verl-omni` pipeline?

File: vllm_omni/diffusion/cache/prompt_embed_cache.py [hunk 2]
```diff
@@ -67,14 +67,44 @@
...
+    except (RuntimeError, TypeError, ValueError, NotImplementedError):
+        return _NOT_HASHABLE
+    return (
+        "__torch_tensor__",
+        str(obj.dtype),
+        tuple(obj.shape),
+        str(obj.device),
```
Comment (SamitHuang) [inline vllm_omni/diffusion/cache/prompt_embed_cache.py:95]: Including `str(obj.device)` means that identical token IDs passed on `cpu` vs `cuda:0` will result in different cache keys and duplicate text encoder executions. Since `encode_prompt` typically returns embeddings on the text encoder's device regardless of the input tensor's device, this device separation reduces the cache hit rate.

File: tests/diffusion/cache/test_prompt_embed_cache.py [hunk 3]
```diff
@@ -363,22 +409,37 @@ def test_different_prompts_miss(self):
...
         assert cache.stats()["misses"] == 2
 
-    def test_tensor_argument_bypasses_and_increments_counter(self):
-        pipe = _FakePipeline()
+    def test_cloned_token_tensors_hit_same_cache_entry(self):
+        pipe = _TensorPromptPipeline()
         cache = install_prompt_embed_cache(pipe)
-        # Put a tensor into a non-precomputed-embed slot to trigger bypass.
-        pipe.encode_prompt("cat", device=torch.device("cpu"), num_images_per_prompt=1)
+        ids = torch.tensor([1, 2, 3], dtype=torch.int64)
```
Comment (SamitHuang) [inline tests/diffusion/cache/test_prompt_embed_cache.py:415]: This test only verifies cache hits for CPU tensors. Add a test case using GPU tensors to verify that accelerator tensors are correctly hashed and do not silently trigger the `_NOT_HASHABLE` exception path.

**Review by SamitHuang — COMMENTED** [review]
This PR adds tensor caching support for `verl-omni`. I have a few concerns regarding the performance impact of the caching mechanism, specifically around device synchronization and key isolation, as well as test coverage for GPU tensors.
